src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs

```diff
@@ -39,20 +38,54 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
             {
                 return Task.FromResult<ITriggerBinding>(null);
             }
-            var resolvedAttribute = GetParameterResolvedAttribute(attribute);
+            var resolvedAttribute = GetParameterResolvedAttribute(attribute, parameterInfo);
             ValidateSignalRTriggerAttributeBinding(resolvedAttribute);
             
             return Task.FromResult<ITriggerBinding>(new SignalRTriggerBinding(parameterInfo, resolvedAttribute, _dispatcher));
         }
 
-        private SignalRTriggerAttribute GetParameterResolvedAttribute(SignalRTriggerAttribute attribute)
+        private SignalRTriggerAttribute GetParameterResolvedAttribute(SignalRTriggerAttribute attribute, ParameterInfo parameterInfo)
         {
             //TODO: AutoResolve more properties in attribute
             var resolvedConnectionString = GetResolvedConnectionString(
                 typeof(SignalRTriggerAttribute).GetProperty(nameof(attribute.ConnectionStringSetting)),
                 attribute.ConnectionStringSetting);
 
-            return new SignalRTriggerAttribute(attribute.HubName, attribute.Category, attribute.Event, attribute.ParameterNames){ConnectionStringSetting = resolvedConnectionString};
+            var hubName = attribute.HubName;
+            var category = attribute.Category;
+            var @event = attribute.Event;
+            var parameterNames = attribute.ParameterNames ?? new string[0]; 
+
+            // We have to model for C#, one is function based model which also work in multiple language
+            // Another one is class based model, which is highly close to SignalR itself but must keep some conventions.
+            var method = (MethodInfo) parameterInfo.Member;
+            var declaredType = method.DeclaringType;
+            // Class based model
+            if (declaredType != null && declaredType.IsSubclassOf(typeof(ServerlessHub)))
+            {
+                hubName = string.IsNullOrEmpty(hubName) ? declaredType.Name : hubName;
+                category = string.IsNullOrEmpty(category) ? GetCategoryFromMethodName(method.Name) : category;
+                @event = string.IsNullOrEmpty(@event) ? method.Name : @event;
```

Comment:
is here we can already detect if both attribute and parameter used and throw exception early, maybe return for non class model as well to save some process below?